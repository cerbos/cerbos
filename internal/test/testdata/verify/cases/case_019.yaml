description: Unspecified effect (https://github.com/cerbos/cerbos/issues/2115)
