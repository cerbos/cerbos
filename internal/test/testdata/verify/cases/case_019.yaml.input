-- example_test.yaml --
name: Example test suite

principals:
  admin:
    id: admin
    roles:
      - admin

resources:
  example:
    kind: example
    id: example

tests:
  - name: Example test
    input:
      principals:
        - admin
      resources:
        - example
      actions:
        - create
        - update
    expected:
      - principal: admin
        resource: example
        actions:
          create: EFFECT_ALLOW
