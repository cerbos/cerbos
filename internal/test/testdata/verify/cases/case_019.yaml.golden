{
  "suites": [
    {
      "file": "example_test.yaml",
      "name": "Example test suite",
      "summary": {
        "overallResult": "RESULT_FAILED",
        "testsCount": 2,
        "resultCounts": [
          {
            "result": "RESULT_PASSED",
            "count": 1
          },
          {
            "result": "RESULT_FAILED",
            "count": 1
          }
        ]
      },
      "testCases": [
        {
          "name": "Example test",
          "principals": [
            {
              "name": "admin",
              "resources": [
                {
                  "name": "example",
                  "actions": [
                    {
                      "name": "create",
                      "details": {
                        "result": "RESULT_FAILED",
                        "failure": {
                          "expected": "EFFECT_ALLOW",
                          "actual": "EFFECT_DENY"
                        }
                      }
                    },
                    {
                      "name": "update",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_DENY"
                        }
                      }
                    }
                  ]
                }
              ]
            }
          ]
        }
      ]
    }
  ],
  "summary": {
    "overallResult": "RESULT_FAILED",
    "testsCount": 2,
    "resultCounts": [
      {
        "result": "RESULT_PASSED",
        "count": 1
      },
      {
        "result": "RESULT_FAILED",
        "count": 1
      }
    ]
  }
}
